pandas/io/tests/test_pickle.py

```diff
@@ -284,6 +284,62 @@ def test_pickle_v0_15_2(self):
         #
         tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
+    def compression_explicit(self, compression):
+        # issue 11666
+        if compression == 'xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path, compression=compression)
+            df2 = pd.read_pickle(path, compression=compression)
+            tm.assert_frame_equal(df, df2)
+
+    def test_compression_explicit(self):
+        compressions = [None, 'gzip', 'bz2', 'xz']
+        for c in compressions:
+            yield self.compression_explicit, c
+
+    def compression_explicit_bad(self, compression):
+        with tm.assertRaisesRegexp(ValueError,
+                                   "Unrecognized compression type"):
+            with tm.ensure_clean(self.path) as path:
+                df = tm.makeDataFrame()
+                df.to_pickle(path, compression=compression)
+
+    def test_compression_explicit_bad(self):
+        compressions = ['', 'None', 'bad', '7z']
+        for c in compressions:
+            yield self.compression_explicit_bad, c
+
+    def compression_infer(self, ext):
+        if ext == '.xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path + ext) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path)
+            tm.assert_frame_equal(df, pd.read_pickle(path))
+
+    def test_compression_infer(self):
+        extensions = ['', '.gz', '.bz2', '.xz', '.who_am_i']
+        for ext in extensions:
+            yield self.compression_infer, ext
+
+    def decompression_prepared_data(self, ext):
```

Comment:
`generate_legacy_store_files.create_data` will generate a `dict`, is there any function which can  compare two `dict` instance? something similar to `tm.assert_frame_equal`?